### ISSUE TITLE
Passing arguments

### ISSUE THREAD

**Issue by Wulfheart** [leaanthony/clir#9]
Is it possible to pass arguments to the cli?
E.g. I have a command like `cli create test` and I'd like the create cmd know that it has to create something named e.g. test.

**Comment by deranjer** [comment]
I'm in the same boat, not sure how to parse the supplied args.

**Comment by leaanthony** [comment]
Clir supports subcommands and named arguments. The [docs](https://clir.leaanthony.com/guide/) have all the examples as well as the examples directory. 

In your example, you'd create a subcommand called `test` and pass in the name using a StringFlag, say `-n`.
EG:
```
package main

import (
	"github.com/leaanthony/clir"
)

func main() {

	// Create new cli
	cli := clir.NewCli("mycli", "My CLI app", "v0.0.1")

	// Add a create subcommand
	create := cli.NewSubCommand("create", "Create a thing")
        thingToCreate := "my default thing"
        create.StringFlag("n", "Name of thing to create", &thingToCreate)
	create.Action(func() error {
		println("I am creating a thing called: " + thingToCreate)
		return nil
	})

	// Run!
	cli.Run()

}
```
Now you can `mycli create -n test`

**Comment by deranjer** [comment]
Yes, we were hoping to do `mycli create test` not `mycli create -n test`.
Or in my example, was hoping to do `mycli add -all` or `mycli add <file>` (kind of like `git add <file>` and `git add --all`)

**Comment by Wulfheart** [comment]
This would be 🔥.

**Comment by deranjer** [comment]
Alright, after some testing have realized we can directly call os.Args[], so as long as we know the position (which could be a major issue), you can fulfill the requirements.  If you don't know the position, then you will need to do some string searching and cutting.

`cli := clir.NewCli("gvcc", "Version control client for GVC", version)
addCmd := cli.NewSubCommand("add", "adds file(s)/folder(s) (recursively for folder) to repo")
//File/Folder/Wildcard adding
addCmd.Action(func() error {
		fmt.Println("File/Folder to add", os.Args[2])
		return nil

})
//Add all files recursively to repo
	addall := addCmd.NewSubCommand("all", "add all of the file(s)/folders(s) recursively to repo")
	addall.Action(func() error {
		fmt.Println("Adding ALL files recursively")
		return nil
})
err := cli.Run()
	if err != nil {
		log.Fatal(err)
	}
}`
`go run client.go add file.txt
File/Folder to add file.txt`

`go run client.go add all
Adding ALL files recursively
`

**Comment by deranjer** [comment]
So maybe something to add to the library that stores all args supplied after the subcommand?  Although all of my actions are functions in another library, so either global or you could implement the changes by amichael7 and their "CustomAction" implementation to pass them, or some other method.

**Comment by leaanthony** [comment]
I'll implement something better than accessing os.Args today :+1:

**Comment by leaanthony** [comment]
I've pushed an update that addresses this. There's an example in `examples/otherags`. I'll update the docs today. Please let me know if this addresses your needs.

**Comment by leaanthony** [comment]
Now tagged as `v1.0.3`.

**Comment by deranjer** [comment]
That was fast! Unfortunately it doesn't work for my use case.  Consider this example:

 ```
func main() {

	// Create new cli
	cli := clir.NewCli("Other Args", "A basic example", "v0.0.1")

	// Set long description
	cli.LongDescription("This app shows positional arguments")

	name := cli.NewSubCommand("name", "Shows your name!")
	name.Action(func() error {
		fmt.Printf("The remaining arguments were: %+v\n", cli.OtherArgs())
		fmt.Println("Show args manually: ", os.Args[1:])
		fmt.Println("Show arg we are interested in: ", os.Args[2])
		return nil
	})
	// Run!
	cli.Run()
}
```
And the output: 
```
>go run test.go name deranjer
The remaining arguments were: []
Show args manually:  [name deranjer]
Show arg we are interested in:  deranjer
```

**Comment by leaanthony** [comment]
Ah, subcommands.... yes.... I can sort that out.

**Comment by deranjer** [comment]
Opened a pull request in case you haven't done it yet.  Was curious if you were planning on adding "required" tags or not to the library?  That is about the only other thing I need at this time.

**Comment by deranjer** [comment]
Hmm, looking at it you wrap the `flagset` from the main library... adding a required tag wouldn't be simple, I'm no expert but you might have to abandon the `flagset` method unless somehow you had a metastructure to add data onto flagset.. well I might mess around with it a little bit.

**Comment by leaanthony** [comment]
To be honest I didn't want to overly complicate the library. It's just meant to be for quick'n'easy clis. Adding required might be tricky. I'm happy to consider any PRs :+1: 

@Wulfheart - does this sort your problem out now?

**Comment by Wulfheart** [comment]
Somehow yes. 

Even though something like a required number of arguments would really be helpful. 😉

**Comment by deranjer** [comment]
@Wulfheart IDK, after some experimentation I think @leaanthony might be right that required is best handled elsewhere.   If you look at your case something like this would be simple and effective:

```
if len(nameCommand.OtherArgs()) == 0 {
	nameCommand.PrintHelp()
	fmt.Println("ERROR: No file/folder name supplied!")
	os.Exit(1)
}
```

This kind of thing requires the flags to be first, but I think that is good practice anyway.  I do have required working on my current git branch, but not sure how much I like it.

However @leaanthony I do have a pull request for shorcuts which I do think are very useful and nice.  Does require a lot of changes though.
